cstr/src/lib.rs

```diff
@@ -59,23 +59,29 @@ pub fn to_widechar_vec(s: &str) -> Vec<WideChar> {
 /// This converts raw C-pointers to rust Strings, which requires unsafe operations
 ///
 #[allow(clippy::uninit_vec)]
-pub unsafe fn input_text_to_string_a(text: *const Char, len: usize) -> String {
-    if (len as isize) < 0 {
-        let mut dst = Vec::new();
-        let mut itr = text;
-        {
-            while *itr != 0 {
-                dst.push(*itr);
-                itr = itr.offset(1);
+pub unsafe fn input_text_to_string_a(text: *const Char, len: isize) -> String {
+    match len {
+        x if x < 0 => {
```

Comment:
I'd prefer if either `x` is changed to `len` or use `x` in the body code over `len`. This code passes the borrow checker because `len` is `Copy`, but would not with something that is not `Copy` (and would require a clone).